pkg/api/types.go

```diff
@@ -708,6 +712,14 @@ type ConfigMapVolumeSource struct {
 	Items []KeyToPath `json:"items,omitempty"`
 }
 
+// DockerVolume represents a volume that is backed by a Docker volume plugin. This depends on the plugin being already properly setup.
```

Comment:
I agree it looks messy, but this might be a less maddening way to deal with NVIDIA applications. They just added, upon my request, more documentation about the internals of their nvidia-docker solution, complete with alternatives that might be more appropriate for projects like Kubernetes (they currently target developers and plain Docker admins):

https://github.com/NVIDIA/nvidia-docker/wiki/NVIDIA-driver

For v0 of the GPU support, we've agreed in node/scheduling meetings not to tie any of the NVIDIA code directly into kubelet and leave most of the burden to users and administrators. The fact that the driver is no longer a strict requirement also explains why I could dedicate fewer cycles to it. This PR would allow additional, incremental experimentation without being tied into the other work. I strongly believe that it's still needed, because the restrictions in V0 are draconian. (We plan to use it.)

For that specific kind of volumes, the cleanup is fine, since it's just removing a tree of hardlinks/copies to system libraries.

I have never even looked at them, but I assume that other kinds of volumes backed by Docker drivers talking to remote resources (Flocker, GlusterFS, Keywhiz, etc.) might be fine with the pod moving, too. I can't claim speaking for those other use cases, though.

If we are not to create/use/destroy (yet), would the expectation be that administrators create in advance all possible NVIDIA volume versions?